gm4_rope_ladders/data/gm4_rope_ladders/functions/ladder_detection/spawn_rcd.mcfunction

```diff
@@ -0,0 +1,7 @@
+# spawns "right click detection" wandering trader
+# @s = player holding rope ladder
+# position of ladder
+# run from gm4_rope_ladders:ladder_detection/found_ladder
+
+summon wandering_trader ~ ~100 ~ {NoGravity:1b,Silent:1b,Invulnerable:1b,NoAI:1b,Tags:["gm4_rol_rcd","gm4_rol_rcd_ladder"],ActiveEffects:[{Id:14b,Amplifier:0b,Duration:999999,ShowParticles:0b}],Offers:{}}
+execute positioned ~ ~100 ~ run tp @e[type=wandering_trader,tag=gm4_rol_rcd_ladder,dx=0] ~ ~-100 ~
```

Comment:
Instead of relying on spatial selection of the `wandering_trader`, you could harden your algorithm by introducing a `new_something` tag, and remove the tag after initialization.